lib/avo/concerns/has_items.rb

```diff
@@ -326,7 +326,7 @@ def is_standalone?(item)
       def hydrate_item(item)
         return unless item.respond_to? :hydrate
 
-        res = self.class.ancestors.include?(Avo::BaseResource) ? self : resource
+        res = (self.class.ancestors.include?(Avo::BaseResource) || self.class.ancestors.include?(Avo::Resources::ArrayResource)) ? self : resource
```

Comment:
let's leave a comment on why the inital behavior was happening.
And abstract it better maybe? Check it out...